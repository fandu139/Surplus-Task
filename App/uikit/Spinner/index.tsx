import Colors from '../../theme/colors';
import React from 'react';
import { ActivityIndicator } from 'react-native';

interface Props {
  size?: 'large' | 'small' | number;
  color?: string;
  testID?: string;
  accessibilityLabel?: string;
}

const Spinner: React.FC<Props> = ({ size, color, testID, accessibilityLabel }: Props) => (
  <ActivityIndicator
    testID={testID}
    accessibilityLabel={accessibilityLabel || testID}
    size={size}
    color={color}
  />
);

Spinner.defaultProps = {
  size: 'large',
  color: Colors.GREEN_47,
  testID: 'loading-indicator',
  accessibilityLabel: undefined,
};

export default Spinner;
