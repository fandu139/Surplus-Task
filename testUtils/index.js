import AuthContext from '../App/context/Auth';
import React from 'react';
import { act, fireEvent, render } from '@testing-library/react-native';

const customRender = (ui, options) => {
  const isAuthenticated = options?.isAuthenticated || false;
  const renderOptions = options?.rendererOptions;
  const authContext = options?.authContext;
  return render(
    <AuthContext.Provider value={{ isAuthenticated, ...authContext }}>{ui}</AuthContext.Provider>,
    renderOptions,
  );
};

// override render method
export { customRender as render };

export function wait(ms) {
  return act(() => new Promise((r) => setTimeout(r, ms)));
}

// Due to debounce on Button UI Kit, we need to add milisecond waiting timer
export function fireAsyncPress(element, ms = 100) {
  fireEvent.press(element);
  return wait(ms);
}
