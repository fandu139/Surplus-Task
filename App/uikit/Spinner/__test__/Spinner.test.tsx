import Spinner from '../';
import React from 'react';
import { render } from '../../../../testUtils';

describe('Spinner component test', () => {
  describe('snapshot test', () => {
    it('should render correctly', () => {
      const tree = render(<Spinner />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
