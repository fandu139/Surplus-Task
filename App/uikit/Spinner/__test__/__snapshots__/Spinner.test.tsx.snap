// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component test snapshot test should render correctly 1`] = `
<ActivityIndicator
  accessibilityLabel="loading-indicator"
  color="#47bba5"
  size="large"
  testID="loading-indicator"
/>
`;
